spotify-api/spotifyRequests.ts

```diff
@@ -206,3 +206,216 @@ export const fetchArtistImagesBySpotifyId = async (
+      }),
+      {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    )
+    .then((response) => {
+      // item limit is 1 above
+      const SpotifyTrackInfo: spotifyYoutubeTrackMapping | undefined = response.data.tracks?.items.map((track) => {
+        return {
+          youtube: {
+            name: playlistTitle.title,
+          },
+          spotify: {
+            uri: track.uri,
+            name: track.name,
+          },
+        };
+      })[0];
+
+      if (SpotifyTrackInfo === undefined) {
+        throw new Error('No spotify track found for video with name ' + playlistTitle.title);
+      }
+
+      return SpotifyTrackInfo;
+    })
+    .catch((err: AxiosError) => {
+      if (err.response?.status === HttpStatusCode.TooManyRequests) {
+        if (currentRetry < maxRetries) {
+          // Retry the request after a delay
+          const retryDelay = parseInt(err.response!.headers['retry-after'] as string) * 1000;
+          //   console.log(`Request failed. Retrying in ${retryDelay / 1000} seconds...`);
+          return new Promise((resolve) => {
+            setTimeout(() => {
+              resolve(
+                fetchSingleSpotifyTrackByYoutubePlaylistURL(accessToken, playlistTitle, maxRetries, currentRetry + 1)
+              );
+            }, retryDelay);
+          });
+        } else {
+          // Max retries reached,
+          // console.log(`Max tries exceeded. Can't get song with title ${playlistTitle.title}`);
+          return {
+            youtube: {
+              name: playlistTitle.title,
+            },
+          };
+        }
+      } else {
+        // Can't find a correspond track for the youtube video
+        // console.log(`Can't get song with title ${playlistTitle.title}`);
+        return {
+          youtube: {
+            name: playlistTitle.title,
+          },
+        };
+      }
+    });
+};
+
+function chunkArray<T>(arr: T[], chunkSize: number) {
+  const result = [];
+  for (let i = 0; i < arr.length; i += chunkSize) {
+    result.push(arr.slice(i, i + chunkSize));
+  }
+  return result;
+}
+
+export const fetchSpotifyTracksByYoutubePlaylistID = (
+  accessToken: string,
+  youtubePlaylistID: string
+): Promise<spotifyYoutubeTrackMapping[]> => {
+  return axios
+    .get<{ id: string; playlistTitles: { title: string }[] }>(
+      queryParamsStringify(`${API_KEY}/playlist`, {
+        id: youtubePlaylistID,
+      })
+    )
+    .then(async (response) => {
+      // batch requests, don't do all at once
+      const BATCH_SIZE = 50;
+      const partitions = chunkArray(response.data.playlistTitles, BATCH_SIZE);
+
+      let spotifyTrackInfoForTitles: spotifyYoutubeTrackMapping[] = [];
+      for (const chunk of partitions) {
+        //  fullfill the 50 promises before going onto the next batch
+        const fetchIntermediateResult = await Promise.all(
+          // todo: Do some preprocessing on the titleOnPlaylist string
```

Comment:
Why not do it now?

Reply:
Its a suggestion; I will change to suggestion.